### ISSUE TITLE
MockedProvider TypeError: Cannot read property 'scheduler' of undefined

### ISSUE THREAD

**Issue by rosskevin** [apollographql/react-apollo#2164]
**Intended outcome:**
Can use `MockedProvider` without error

**Actual outcome:**
```
Cannot read property 'scheduler' of undefined
            TypeError: Cannot read property 'scheduler' of undefined
    at MockedProvider../node_modules/react-apollo/test-utils.js.MockedProvider.componentWillUnmount (http://localhost:6006/static/preview.bundle.js:120042:60)
    at callComponentWillUnmountWithTimer (http://localhost:6006/static/preview.bundle.js:137137:12)
```

**How to reproduce the issue:**
Use a `MockedProvider` without any provided mocks, the queryManage may be undefined it appears.

![storybook](https://user-images.githubusercontent.com/136564/42346873-a5aa794e-8069-11e8-9fb2-1fb8e4f5b5f7.png)

**Version**

* apollo-client@2.3.5
* react-apollo@2.1.9

Introduced in #2151 by @danilobuerger and merged by myself.  First release was 2.1.9.

I am newly using this in my code, strange that it isn't surfaced in our own tests.  Investigating now.

**Comment by danilobuerger** [comment]
Thats weird, I thought I also tested for this. This should also cause a bug in apollo-client as the typings are wrong in that case. Do you want me to fix it or are you already working on it?

**Comment by rosskevin** [comment]
How about you PR a null check and I'll investigate why this is occurring.  So far at least the type in ApolloClient indicates it is not undefined.

**Comment by danilobuerger** [comment]
So I can reproduce this whenever there is no graphql inside MockedProvider.

**Comment by danilobuerger** [comment]
Which makes sense because the query manager is only ever initialized with: https://github.com/apollographql/apollo-client/blob/master/packages/apollo-client/src/ApolloClient.ts#L373
after a start query is called

**Comment by rosskevin** [comment]
Yes, exactly - it is happening in my storybooks where the provider is setup but no query gets executed.  The queryManager is null until something is executed, so the type is wrong in apollo-client.

**Comment by danilobuerger** [comment]
So I would do a PR on apollo-client that this could be undefined and follow up here with a null check.

**Comment by rosskevin** [comment]
Perfect, thank you very much for that!

**Comment by danilobuerger** [comment]
Fixed in #2165
